extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java

```diff
@@ -306,6 +307,19 @@ public static <K, V> boolean isNullOrEmpty(final Map<K, V> map) {
         return map == null || map.isEmpty();
     }
 
+    /**
+     * Checks whether a {@link Page} is null or empty.
+     * <br>
+     * The page is empty when the page's URL is null or empty and the page's body is null.
+     *
+     *  @param page the {@link Page} to check
+     * @return whether the page is empty when the page's URL is null or empty
+     * and the page's body is null.
+     */
+    public static boolean isNullOrEmpty(final Page page) {
+        return page == null || (isNullOrEmpty(page.getUrl()) && page.getBody() == null);
+    }
```

Comment:
We already have `Page.isValid`, so you might want to change that to be
```
return page != null && (
    !isNullOrEmpty(page.getUrl())
    || !isNullOrEmpty(page.getIds())
    || page.getBody() != null);
```

Reply:
The logic of your posted code does not seem right to me. When the page is `null`, the method will always return `false`;
I'd go with
``` Java
return page == null || (
        isNullOrEmpty(page.getUrl())
                && isNullOrEmpty(page.getIds())
                && page.getBody() == null
);
```

Reply:
Yes, that's because I'm talking about the `Page.isValid(Page page)` function, which already exists and whose logic is inverted. This function should be removed in favour of that one.

Reply:
Ah ok. Yeah, that makes sense :+1:

Reply:
Silly me did not check whether the method is still used...
I needed it for a previous implementation. The `isNullOrEmpty(final Page page)` is not used at all. I just removed the method and that should be enough. A page should still have some kind of identifying info in it.